src/learning/dbscan.rs

```diff
@@ -178,11 +178,8 @@ impl DBSCAN {
             let visited = self._visited[*data_point_idx];
             if !visited {
                 self._visited[*data_point_idx] = true;
-                let sub_neighbours =
-                    self.region_query(&inputs.data()[data_point_idx * inputs.cols()..(data_point_idx +
-                                                                                   1) *
-                                                                                  inputs.cols()],
-                                      inputs);
+                let sub_neighbours = self.region_query(inputs.select_rows(&[*data_point_idx]).data(),
```

Comment:
I'm inclined to keep this as is until https://github.com/AtheMathmo/rulinalg/issues/7 has been tackled. I'll add a comment on that ticket to note that this piece of code should be revisited once that feature lands.

Happy to hear your thoughts on it.